ietf-optical-impairment-topology.yang

```diff
@@ -876,57 +876,47 @@ module ietf-optical-impairment-topology {
     description "OTSiG definition , representing client
      digital information stream supported by 1 or more OTSi";
 
-     list otsi {
-       key "otsi-carrier-id";
-       config false;
-       description
-         "list of OTSi contained in 1 OTSiG.
+    list otsi {
+      key "otsi-carrier-id";
+      config false;
+      description
+        "list of OTSi contained in 1 OTSiG.
         The list could also be of only 1 element";
-       leaf otsi-carrier-id {
-         type int16;
-         description "OTSi carrier-id";
-       }
+      leaf otsi-carrier-id {
+        type int16;
+        description "OTSi carrier-id";
+      }
 
 /*any OTSi as signal generated by transceiver and*/
 /* attached to a transponder.*/ 
        
     
-       leaf transponder-ref { 
-            type leafref {
-              path "/nw:networks/nw:network/nw:node/tet:te" +
-                "/tet:tunnel-termination-point" +
-                "/transponder/transponder-id";
-            }
-          description
-              "Reference to the configured transponder";
-       }
-       leaf transceiver-ref {
-            type leafref {
-              path "/nw:networks/nw:network/nw:node/tet:te" +
-                "/tet:tunnel-termination-point/"  
-               +"transponder[transponder-id=current()"
-               +"/../transponder-ref]/"
-               + "transceiver/transceiver-id" ;
-              }
-          description
-             "Reference to the configured transceiver " ;
-       }
-       leaf configured-mode { 
-         type leafref {
-           path "/nw:networks/nw:network/nw:node/tet:te" +
-                "/tet:tunnel-termination-point/"  
-               +"transponder[transponder-id=current()"
-               +"/../transponder-ref]/"+
-               "transceiver[transceiver-id=current()/"+
-               "../transceiver-ref]/supported-modes/"+
+      leaf transponder-ref { 
+        type leafref {
+          path "../../../../../transponder/transponder-id";
```

Comment:
> Why is this changed to relative path ?

We have changed the leafref from absolute to relative to be sure to stay on the correct "network" , since we have a list of network in the xpath related to RFC 8345. Using absolute path , while it can be used in any schema tree, does not assure the correct scope. On the other hand , as mentioned in issue #79 , in case would exist a way to "constraint" the scope, the absolute xpath would be better, since using the relative path in a grouping would not be correct unless you know in advance where that grouping would be used.